c++/src/kj/string.h

```diff
@@ -133,11 +133,15 @@ class StringPtr {
   inline constexpr const char* end() const { return content.end() - 1; }
 
   inline constexpr bool operator==(decltype(nullptr)) const { return content.size() <= 1; }
+  inline constexpr bool operator<=(decltype(nullptr)) const { return *this == nullptr; }
+  inline constexpr bool operator>=(decltype(nullptr)) const { return true; }
+  inline constexpr bool operator< (decltype(nullptr)) const { return false; }
+  inline constexpr bool operator> (decltype(nullptr)) const { return content.size() > 1; }
 
-  inline bool operator==(const StringPtr& other) const;
-  inline bool operator< (const StringPtr& other) const;
+  inline bool operator==(const StringPtr& other) const { return content.first(content.size() - 1) == other.content.first(other.content.size() - 1); }
```

Comment:
you're delegating comparison to ArrayPtr, right? Can you use asBytes() or something?

Reply:
Why? either way it invokes the same memcmp? is there some benefit for using asBytes?